includes/template-library/manager.php

```diff
@@ -422,6 +423,16 @@ public function export_template( array $args ) {
 			return $validate_args;
 		}
 
+		$post_id = intval( $args['template_id'] );
+		$post = get_post( $post_id );
+		if ( ! $post || get_post_type( $post ) !== Source_Local::CPT ) {
+			return new \WP_Error( 'template_error', esc_html__( 'Invalid template type or template does not exist', 'elementor' ) );
+		}
+
+		if ( 'private' === $post->post_status && ! current_user_can( 'read_private_posts', $post_id ) ) {
```

Comment:
I think get_post is redundant here. get_post_type will return false if post does not exist.
```suggestion
		$post_status = get_post_status( $post_id );
		if ( get_post_type( $post ) !== Source_Local::CPT ) {
			return new \WP_Error( 'template_error', esc_html__( 'Invalid template type or template does not exist', 'elementor' ) );
		}

		if ( 'private' === $post_status && ! current_user_can( 'read_private_posts', $post_id ) ) {
```

Reply:
Also currently the code only checks for "private". This does not cover the scenario in which a lower user role (e.g. Author) can export a "draft" template not created by the user. In this case the check should be for post_status of "draft" and capability of "edit_post".
A bit of a different approach can be checking any status that is NOT "publish".